### ISSUE TITLE
3.0.0: @property tag showing in class Tags list and other nit-picks

### ISSUE THREAD

**Issue by jrfnl** [phpDocumentor/phpDocumentor#2439]
Similar/related to #2438.

**Given the following piece of code in file `src/issue-2439.php`:**
``` php
<?php
/**
 * Class summary.
 *
 * @property       int    $foo Description of foo.
 * @property-read  bool   $bar Description of bar.
 * @property-write string $baz Description of baz.
 *                             Extended description of baz.
 */
class Issue2439
{
    /**
     * Storage for undeclared properties.
     */
    private $customProps = [];

    /**
     * Getter.
     *
     * @param string $name Property name.
     *
     * @return mixed
     */
    public function __get($name)
    {
        switch ($name) {
            case 'foo':
            case 'bar':
                if (isset($this->customProps[$name])) {
                    return $this->customProps[$name];
                }

                return null;
        }

        throw new Exception("Property $name can not be read");
    }

    /**
     * Setter.
     *
     * @param string $name  Property name.
     * @param mixed  $value New property value.
     *
     * @return mixed
     */
    public function __set($name, $value)
    {
        switch ($name) {
            case 'foo':
            case 'baz':
                $this->customProps[$name] = $value;
                break;
        }

        throw new Exception("Property $name can not be written to");
    }
}
```

<details>
  <summary><b>with the following config:</b></summary>

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<phpdocumentor
    configVersion="3"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns="https://www.phpdoc.org"
    xsi:noNamespaceSchemaLocation="https://docs.phpdoc.org/latest/phpdoc.xsd"
>
    <title>Demo</title>
    <version number="latest">
        <api format="php">
            <source dsn=".">
                <path>src</path>
            </source>
        </api>
    </version>
    <template name="default"/>
</phpdocumentor>
```
</details>

## Expected behavior
I'd expect:

TOC:
1. The properties to show in the class TOC. :heavy_check_mark:
2. The properties added via the `@property(-[read|write])` tag to be listed below "normal" properties. :x:
3. The `property` tags not to show in the class Tags list. :x:
3. The description to show as the property summary if one sentence (up to the first period). If more than one sentence: the first sentence to show as the property summary. :x:

Properties list:
1. The properties to show in the Properties list :heavy_check_mark:
2. Read-only/write-only properties to be indicated as such. Either with an annotation after the property name (see screenshot) or by adding a `property-read` or `read-only` / `property-write` or `write-only` tag to the Tags list for the property. :x:
3. The first sentence of the description (up to the first period) to show as the method summary. :x:
4. The rest of the description to show as the long description. :x:

## Actual behavior
See the :heavy_check_mark: and :x:'s above.

The more serious issue is the property tag showing in the class Tags list (TOC:3).

I'm not sure if this is a templating issue or an issue with the way the data is saved in the AST cache, but the way it is now, is confusing and more noise than having the tag is worth.

Having a "read/write-only" indication is second on my priority list.

The list order is third on my priority list.

The summary being shown as description is something I can live with if needs be.

### TOC
> ![issue-2439-class-overview](https://user-images.githubusercontent.com/663378/84738847-e9cc4180-afaa-11ea-8e62-e925c8e5796a.png)

### Properties list
> ![issue-2439-property-overview](https://user-images.githubusercontent.com/663378/84738851-eafd6e80-afaa-11ea-92ae-a4844024c7fc.png)

## Steps to reproduce the problem

1. Issue can be reproduced with the above code sample and config.
2. Run `php -f /path/to/phpDocumentor/bin/phpdoc`

## Your environment

* Version used: phpDocumentor v3.0.x-dev@d15f2d6
* Install method: git clone
* PHP version: 7.4.6
* Operating system and version: Windows 7

**Comment by mvriel** [comment]
Resolved by 28d82cd

The only thing that is not addressed is to show the first sentence of the description as summary change as this is an interesting bit and I will create a separate improvement for it

**Comment by jrfnl** [comment]
@mvriel I can see that the issue is addressed, however, this commit broke other things... see #2462
